resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -463,6 +466,20 @@ public DnsNameResolverBuilder decodeIdn(boolean decodeIdn) {
         return this;
     }
 
+    /**
+     * Set the maximum numbers of lookups that we try to consolidate when in-flight. This means if multiple
+     * lookups are done for the same hostname and still in-flight only one actual query will be made and the
+     * result will be cascaded to the others.
+     *
+     * @param maxNumConsolidation the maximum lookups to consolidate, or {@code 0} if no consolidation should be
+     *                            performed.
+     * @return {@code this}
+     */
+    public DnsNameResolverBuilder consolidateLookups(int maxNumConsolidation) {
```

Comment:
If the consolidation was built into the main cache itself, it would operate like a read-through cache, and this parameter would be the cache size.

What's not clear from these docs is that this is the number of hostnames. Not the number of concurrent lookups, nor the number of lookups-per-hostname. The latter two have no limit, AFAIK.

I would call this the "consolidation cache size" and explain that it's the maximum number of distinct hostnames for which concurrent queries will be deduplicated and consolidated.

Reply:
maybe `consolidatePerHostnameLookups` ?

Reply:
Agreed with Chris, his proposal makes the behavior clear.
`consolidatePerHostnameLookups` is still misleading, it makes users feel that the property controls number concurrent lookups per hostname, not the total number of distinct hostnames.

Reply:
any naming suggestions ? I am still having a hard time to come up with a good name :/

Reply:
`consolidationCacheSize`?

Reply:
not sure if its really a "cache" tho... But I guess its better then the current name, so lets use it